### PR TITLE
Move Control onto derived RecordType

### PR DESCRIPTION
**Pull request by lesaltzm**
The DKind.Control is a special case of a record. Now that we support derived record types, we can eliminate this special case and move it to be an impl of RecordType.

This PR picks up the PowerFx side of the changes needed to make that work.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Types/DType.cs [hunk 1]
```diff
@@ -3197,10 +3171,33 @@ public virtual bool CoercesTo(DType typeDest, out bool isSafe, out DType coercio
             return doesCoerce;
         }
 
-        protected virtual bool? SubtypeCoercesTo(DType typeDest, ref bool isSafe, out DType coercionType, ref KeyValuePair<string, DType> schemaDifference, ref DType schemaDifferenceType)
+        // This is a specific function to support PrimaryOutput Properties on Control LazyRecord types
+        // If we ever support DefaultField output for Records more broadly, this would be where it's done.
+        private bool? SubtypeCoercesTo(DType typeDest, ref bool isSafe, out DType coercionType, ref KeyValuePair<string, DType> schemaDifference, ref DType schemaDifferenceType)
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Core/Types/DType.cs:3176]: add SubtypeCoercesTo tests in PowerFx repo

File: src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs [hunk 2]
```diff
@@ -20,6 +20,9 @@ public abstract class AggregateType : FormulaType
         /// </summary>
         public virtual string UserVisibleTypeName => null; 
 
+        // Internal property, allows Canvas to control whether a Union operation is permitted on a derived record type
+        internal virtual bool AllowsUnion => true;
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs:24]: May want to make this public in the future. Applies to cases like
`If(x, DerivedRecord1, DerivedRecord2)` and controls whether that will type check or not. We may want to allow derived record types beyond controls to restrict this.

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by lesaltzm** [comment]
Closed as a bit too risky at the moment. #598 picks up the generally useful learnings from this exploration.

**Pull request closed without merging**